test/Microsoft.ML.Tests/TextLoaderTests.cs

```diff
@@ -840,6 +840,302 @@ public void TestTextLoaderBackCompat_VerWritt_0x0001000C()
+            string dataPathDecimalMarkerComma = GetDataPath("iris-decimal-marker-as-comma.txt");
+            var readerDecimalMarkerPeriod = new TextLoader(mlContext, new TextLoader.Options()
+            {
+                Columns = new[]
+                        {
+                            new TextLoader.Column("Label", DataKind.UInt32, 0),
+                            new TextLoader.Column("Features", DataKind.Single, new [] { new TextLoader.Range(1, 4) }),
+                        },
+                DecimalMarker = '.'
+            });
+            var textDataMismatched2 = readerDecimalMarkerPeriod.Load(GetDataPath(dataPathDecimalMarkerComma));
+
+            DataViewSchema columnsComma = textDataMismatched2.Schema;
+            using DataViewRowCursor cursorComma = textDataMismatched2.GetRowCursor(columnsComma);
+            VBuffer<Single> featuresComma = default;
+            ValueGetter<VBuffer<Single>> featuresDelegateComma = cursorComma.GetGetter<VBuffer<Single>>(columnsComma[1]);
+
+            // Iterate over each row
+            while (cursorComma.MoveNext())
+            {
+                featuresDelegateComma.Invoke(ref featuresComma);
+                foreach (float feature in featuresComma.GetValues().ToArray())
+                    Assert.Equal(feature, Single.NaN);
+            }
+        }
+
+        [Fact]
+        public void TestCommaAsDecimalMarkerWithSeperatorAsCommaInCSV()
+        {
+            // Check to confirm TextLoader can read data from a CSV file where the separator is ',' and decimals
+            // enclosed with quotes and with the decimal marker ',' can be successfully read.
+            string dataPathCsv = GetDataPath("iris-decimal-marker-as-comma.csv");
+
+            var mlContext = new MLContext(seed: 1);
+            var readerCsv = new TextLoader(mlContext, new TextLoader.Options()
+            {
+                Columns = new[]
+                        {
+                            new TextLoader.Column("Label", DataKind.Single, 0),
+                            new TextLoader.Column("Features", DataKind.Single, new [] { new TextLoader.Range(1, 4) }),
+                        },
+                DecimalMarker = ',',
+                Separator = ",",
+                AllowQuoting = true,
+                HasHeader = true
+            });
+            var textDataCsv = readerCsv.Load(GetDataPath(dataPathCsv));
+
+            // Load values from iris-decimal-marker-as-comma.csv
+            DataViewSchema columnsCsv = textDataCsv.Schema;
+            using DataViewRowCursor cursorCsv = textDataCsv.GetRowCursor(columnsCsv);
+            UInt32 labelCsv = default;
+            ValueGetter<UInt32> labelDelegatePeriod = cursorCsv.GetGetter<UInt32>(columnsCsv[0]);
+            VBuffer<Double> featuresCsv = default;
+            ValueGetter<VBuffer<Double>> featuresDelegatePeriod = cursorCsv.GetGetter<VBuffer<Double>>(columnsCsv[1]);
```

Comment:
Your getters and the objects where you store values should be Single, since you're reading them as Single.

Reply:
Yup already did that change.